### DISCUSSION TITLE
How to get slanting lines in mini.statusline and mini.tabline like in the below picture.

### DISCUSSION THREAD

**Discussion by 231tr0n** [nvim-mini/mini.nvim#1181] [Q&A]
![image](https://github.com/user-attachments/assets/16593171-f7a1-4e0d-ba31-0cc5255a2807)
Hi, Can someone please help me figure out how can I get a statusline like this with slanting lines before and after each module in mini.statusline.
Also, can the same be done for mini.tabline as well.

**Comment by echasnovski** [comment]
This is done by manually placing triangular looking shapes in proper places and highlight them accordingly. The shapes usually are , , or 🭋, 🭀, 🭛, 🭦. See [list of box drawing characters](https://en.wikipedia.org/wiki/Box-drawing_characters) for more. However, this would require manual construction of statusline from parts (without the help from [`combine_groups`](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-statusline.txt#L168)). So something like this:

```lua
local content_active = function()
  local mode, mode_hl = MiniStatusline.section_mode({ trunc_width = 120 })
  local parts = { '%#', mode_hl, '# ', mode, '🭦', '%#StatusLine#' }
  return table.concat(parts, '')
end
require('mini.statusline').setup({ content = { active = content_active } })
```

The rest of parts are up to you to decide how you want them to look. Here is [an example of default content](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-statusline.txt#L86-L106) you can use for reference. Note that this might require defining custom highlight groups for those edges slanted specifically.

------

The same idea (manually pad with triangular-like shapes) goes for 'mini.tabline' utilizing [`config.format`](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-tabline.txt#L107-L120).

------

However, the *real* answer here is that both 'mini.statusline' and 'mini.tabline' are *intentionally* designed to have vertical separators. Those to me look the cleanest and most space efficient. If you have to customize them to the point when it requires a lot of tweaking and code, then they probably are not for you and looking at other statusline/tabline plugins is a wiser choice.

**Reply by 231tr0n** [reply]
> This is done by manually placing triangular looking shapes in proper places and highlight them accordingly. The shapes usually are , , or 🭋, 🭀, 🭛, 🭦. See [list of box drawing characters](https://en.wikipedia.org/wiki/Box-drawing_characters) for more. However, this would require manual construction of statusline from parts (without the help from [`combine_groups`](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-statusline.txt#L168)). So something like this:
> 
> ```lua
> local content_active = function()
>   local mode, mode_hl = MiniStatusline.section_mode({ trunc_width = 120 })
>   local parts = { '%#', mode_hl, '# ', mode, '🭦', '%#StatusLine#' }
>   return table.concat(parts, '')
> end
> require('mini.statusline').setup({ content = { active = content_active } })
> ```
> 
> The rest of parts are up to you to decide how you want them to look. Here is [an example of default content](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-statusline.txt#L86-L106) you can use for reference. Note that this might require defining custom highlight groups for those edges slanted specifically.
> 
> ------
> 
> The same idea (manually pad with triangular-like shapes) goes for 'mini.tabline' utilizing [`config.format`](https://github.com/echasnovski/mini.nvim/blob/b5ee41d2b8ab54f8b6b383b488e822c5cc431871/doc/mini-tabline.txt#L107-L120).
> 
> ------
> 
> However, the *real* answer here is that both 'mini.statusline' and 'mini.tabline' are *intentionally* designed to have vertical separators. Those to me look the cleanest and most space efficient. If you have to customize them to the point when it requires a lot of tweaking and code, then they probably are not for you and looking at other statusline/tabline plugins is a wiser choice.

Thank you soo much for helping me out with this. Your plugins are the best. 🤘🤟
Also for the last part of the answer I believe if it is possible, then they are built for it since I realised all I needed were the box characters.
